linkerd/protocol/http/src/main/scala/io/buoyant/linkerd/protocol/HttpConfig.scala

```diff
@@ -31,6 +31,7 @@ class HttpInitializer extends ProtocolInitializer.Simple {
       .prepend(Headers.Dst.PathFilter.module)
       .replace(StackClient.Role.prepFactory, DelayedRelease.module)
       .prepend(http.ErrorResponder.module)
+      .insertBefore(ClassifiedRetries.role, RequestFramingFilter.module)
```

Comment:
I think this can move even higher in the path stack.  Since the request is invalid, we want to reject it as quickly as possible so that we don't waste resources processing it.  I would say you could just prepend it here to put it at the top of the stack.  Or maybe right below ErrorResponder just in case this filter ever throws an exception (even though it shouldn't).